pxd.c

```diff
@@ -1915,16 +1921,54 @@ static void pxd_root_dev_release(struct device *dev)
 {
 }
 
+static ssize_t pxd_num_fpthreads_show(struct device *dev,
+			 struct device_attribute *attr, char *buf)
+{
+	return sprintf(buf, "%u\n", pxd_num_fpthreads);
+}
+
+static ssize_t pxd_num_fpthreads_store(struct device *dev, struct device_attribute *attr,
+			   const char *buf, size_t count)
+{
+	uint32_t new_pxd_num_fpthreads = 0;
+
+	sscanf(buf, "%u", &new_pxd_num_fpthreads);
+
+	mutex_lock(&sysfs_lock);
+	fastpath_adjust_fpthreads(new_pxd_num_fpthreads);
+	mutex_unlock(&sysfs_lock);
+
+	return count;
+}
+
+static DEVICE_ATTR(pxd_num_fpthreads, S_IRUGO|S_IWUSR, pxd_num_fpthreads_show, pxd_num_fpthreads_store);
+
+static struct attribute *pxd_root_attrs[] = {
+	&dev_attr_pxd_num_fpthreads.attr,
+	NULL
+};
+
+static struct attribute_group pxd_root_attr_group = {
+	.attrs = pxd_root_attrs,
+};
+
+static const struct attribute_group *pxd_root_attr_groups[] = {
+	&pxd_root_attr_group,
+	NULL
+};
+
+static struct device_type pxd_root_device_type = {
+	.name		= "pxd",
+	.groups		= pxd_root_attr_groups,
+	.release	= pxd_root_dev_release,
+};
+
 static struct device pxd_root_dev = {
 	.init_name =    "pxd",
 	.release =      pxd_root_dev_release,
+	.type = &pxd_root_device_type,
```

Comment:
since you introduced at the top level to configure pxd devices - which is absolutely the correct thing to do...
unrelated but relevant:
can you bring pxd_timeout_show/store also here and control the pxd_timeout_msecs.
Currently this configuration is possible only if a pxd device exists through its minor dev path. - very ugly.